### PR TITLE
Adapt FlexibleSpaceBar to iOS

### PR DESCRIPTION
**Pull request by abarth**
We should center the title of flexible space bars on iOS.

Related to #4962

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @HansMuller

File: packages/flutter/lib/src/material/flexible_space_bar.dart [hunk 1]
```diff
@@ -63,6 +73,19 @@ class _FlexibleSpaceBarState extends State<FlexibleSpaceBar> {
     super.deactivate();
   }
 
+  bool _getEffectiveCenterTitle(ThemeData theme) {
+    if (config.centerTitle != null)
+      return config.centerTitle;
+    assert(theme.platform != null);
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/flexible_space_bar.dart]: Couldn't we just allow this to be null?

File: packages/flutter/lib/src/animation/animations.dart [hunk 2]
```diff
@@ -388,7 +390,19 @@ class CurvedAnimation extends Animation<double> with AnimationWithParentMixin<do
     if (activeCurve == null)
       return t;
     if (t == 0.0 || t == 1.0) {
-      assert(activeCurve.transform(t).round() == t);
+      assert(() {
+        final double transformedValue = activeCurve.transform(t);
+        final double roundedTransformedValue = transformedValue.round().toDouble();
+        if (roundedTransformedValue != t) {
+          throw new FlutterError(
```
Comment (HansMuller) [inline packages/flutter/lib/src/animation/animations.dart]: NICE

File: packages/flutter/lib/src/material/flexible_space_bar.dart [hunk 3]
```diff
@@ -101,6 +124,7 @@ class _FlexibleSpaceBarState extends State<FlexibleSpaceBar> {
 
     // title
     if (config.title != null) {
+      final ThemeData theme = Theme.of(context);
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/flexible_space_bar.dart]: We call this variable themeData about 50% of the time. I like themeData a little better :-)

**Comment by HansMuller** [comment]
LGTM

File: packages/flutter/lib/src/material/flexible_space_bar.dart (hunk 3 above)
Comment (abarth) [inline packages/flutter/lib/src/material/flexible_space_bar.dart]: I'm pretty sure there's a bug on file for this issue already.  :)

File: packages/flutter/lib/src/animation/animations.dart [hunk 4]
```diff
@@ -388,7 +390,19 @@ class CurvedAnimation extends Animation<double> with AnimationWithParentMixin<do
...
+            'Invalided curve endpoint at $t.\n'
```
Comment (Hixie) [inline packages/flutter/lib/src/animation/animations.dart]: typo in "Invalid"

**Pull request merged**